src/devices/video/epic12.cpp

```diff
@@ -769,81 +778,50 @@ void epic12_device::gfx_exec_w(address_space &space, offs_t offset, u32 data, u3
 				osd_work_item_release(m_blitter_request);
 			}
 
-			blit_delay = 0;
-			gfx_create_shadow_copy(space); // create a copy of the blit list so we can safely thread it.
-
-			if (blit_delay)
-			{
-				m_blitter_busy = 1;
-				m_blitter_delay_timer->adjust(attotime::from_nsec(blit_delay*8)); // NOT accurate timing (currently ignored anyway)
-			}
-
-			m_gfx_addr_shadowcopy = m_gfx_addr;
 			m_gfx_scroll_0_x_shadowcopy = m_gfx_scroll_0_x;
 			m_gfx_scroll_0_y_shadowcopy = m_gfx_scroll_0_y;
 			m_gfx_scroll_1_x_shadowcopy = m_gfx_scroll_1_x;
 			m_gfx_scroll_1_y_shadowcopy = m_gfx_scroll_1_y;
-			m_blitter_request = osd_work_item_queue(m_work_queue, blit_request_callback, (void*)this, 0);
-			//g_profiler.stop();
-		}
-	}
-}
 
+			// Create a copy of the blit list so we can safely thread it.
+			// Copying the Blitter operations will also estimate the delay needed for processing.
+			m_blit_delay_ns = 0;
+			gfx_create_shadow_copy(space); 
 
-void epic12_device::gfx_exec_w_unsafe(offs_t offset, u32 data, u32 mem_mask)
-{
-	if (ACCESSING_BITS_0_7)
-	{
-		if (data & 1)
-		{
-			//g_profiler.start(PROFILER_USER1);
-			// make sure we've not already got a request running
-			if (m_blitter_request)
-			{
-				int result;
-				do
-				{
-					result = osd_work_item_wait(m_blitter_request, 1000);
-				} while (result == 0);
-				osd_work_item_release(m_blitter_request);
-			}
+			// Every EP1C_VRAM_H_LINE_PERIOD_NANOSEC, the Blitter will block other operations, due
+			// to fetching a horizontal line from VRAM for output.
+			m_blit_delay_ns += std::floor( m_blit_delay_ns / EP1C_VRAM_H_LINE_PERIOD_NANOSEC ) * EP1C_VRAM_H_LINE_DURATION_NANOSEC;
 
-			if (blit_delay)
-			{
-				m_blitter_busy = 1;
-				int delay = blit_delay*(15 * m_delay_scale / 50);
-				//printf("delay %d\n", delay);
-				m_blitter_delay_timer->adjust(attotime::from_nsec(delay));
-			}
-			else
-			{
-				m_blitter_busy = 0;
-			}
+			#if EP1C_PRINT_BLITTER_DELAYS
+				// Check if Blitter takes longer than a frame to render.
+				// In practice, there's a bit less time than this to allow for lack of slowdown but
+				// for debugging purposes this is an ok approximation.
+				if (m_blit_delay_ns > EP1C_FRAME_DURATION_NANOSEC)
+					printf("Blitter delay! Blit duration %lld ns.\n", m_blit_delay_ns);
+			#endif
```

Comment:
Please avoid using the preprocessor like this – the logging code will rot.  You can use the configurable logging from logmacro.h (the compiler is smart enough to optimise it away if logging is disabled).

Reply:
I think the way I changed it to is correct?
Please verify. I'm not used to Mame libraries.